src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,30 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
-
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+      Animal.alive = Animal.alive.filter(alive => alive.health > 0);
```

Comment:
Ok, now it's a bit better, but still this array will be filter even if animal is not dead yet 🙂 Just check if this animal has 0 or less health and perform filter only in that case